src/hittable.rs

```diff
@@ -236,23 +245,23 @@ aarect!(YZRect, y, z, normal: x);
 
 
 #[derive(Debug)]
-pub struct Stage {
+pub struct HittableList {
     objects: Vec<Box<dyn Hittable>>,
     aabb: AABB,
 }
 
-impl Stage {
-    pub fn new(objects: Vec<Box<dyn Hittable>>) -> Stage {
+impl HittableList {
+    pub fn new(objects: Vec<Box<dyn Hittable>>) -> HittableList {
         let aabb = (|| {
             let mut aabbs = objects.iter().flat_map(|o| o.bounding_box(0.0, 1.0));
             let first = aabbs.next()?;
             Some(aabbs.fold(first, |a, b| a + b))
         })();
```

Comment:
Since you `.unwrap`s `aabb` afterwards anyway, you can simplify it to:
```
let aabb = {
    let first = aabbs.next().unwrap();
    aabbs.fold(first, |a, b| a + b)
};
```